Automattic/wp-calypso
# QR Code Login: Start Requesting the QR Code data and auth url.

#### Changes proposed in this Pull Request
In this PR the QR Code gets generated when you visit /log-in/qr based on the token retrieved by the new login actions—defined in D79766-code.

This PR implements the browser side of the pdjUbD-db-p2 QR Code Login project
By pulling the QR auth endpoint for the login url ( magic link) 

See
<img width="600" src="https://user-images.githubusercontent.com/115071/167772947-38a9c302-8edd-4265-8022-0cc695c92ec1.png" />

#### Testing instructions
1. Apply D79766-code to you .com sandbox.
2. and sandbox wordpress.com and public-api.wordpress.com to point to you sandbox. 
3. Logout of WordPress.com or use an incognito browser. (Also please test across different browsers and devices)
By running this PR locally or by using the Calypso live link you should be able to visit. (calypso-url)/log-in and
navigate to /log-in/qr via a link. 

A) Notice that the QR code loads. (Scanning the code using the device currently opens up the WordPress App or sends you to install it. 
B) Open the network tab in the devloper console and notice the "polling" of the auth endpoint. 
Also notice that the if you change the expiration to 30(sec) in D79766-code (Line 25 in wp-content/lib/qr-login/class-wpcom-qr-login.php) that the QR code updates to new one. 
C) Notice that the QR code persists when you refresh the page. 

Related to D79766-code, https://github.com/Automattic/wp-calypso/pull/63231
